eip_operator/src/controller/node.rs

```diff
@@ -157,3 +186,139 @@ impl k8s_controller::Context for Context {
         Ok(None)
     }
 }
+
+/// Applies label to node specifying the status of the egress gateway node.
+#[instrument(skip(api), err)]
+async fn add_gateway_status_label(
+    api: &Api<Node>,
+    name: &str,
+    status: &str,
+) -> Result<Node, kube::Error> {
+    info!(
+        "Adding gateway status label {} key {} to node {}",
+        crate::EGRESS_NODE_STATUS_LABEL,
+        status,
+        name
+    );
+    let patch = serde_json::json!({
+        "apiVersion": "v1",
+        "kind": "Node",
+        "metadata": {
+            "labels": {
+                // Ensure the status is lowercase to match conventions
+                crate::EGRESS_NODE_STATUS_LABEL: status.to_lowercase(),
+            }
+        }
+    });
+    let patch = Patch::Apply(&patch);
+    let params = PatchParams::apply(crate::FIELD_MANAGER);
+    api.patch(name, &params, &patch).await
+}
+
+/// Retrieve all egress nodes in the cluster.
+async fn get_egress_nodes(api: &Api<Node>) -> Result<Vec<Node>, kube::Error> {
+    let params = ListParams::default().labels(
+        format!(
+            "{}={}",
+            EGRESS_GATEWAY_NODE_SELECTOR_LABEL_KEY, EGRESS_GATEWAY_NODE_SELECTOR_LABEL_VALUE
+        )
+        .as_str(),
+    );
+
+    match api.list(&params).await {
+        Ok(node_list) => Ok(node_list.items),
+        Err(e) => Err(e),
+    }
+}
+
+/// Get Ready status from the node status field.
+async fn get_ready_status_from_node(node: &Node) -> Option<String> {
```

Comment:
this doesn't need to be async

Reply:
Fixed in [commit ](https://github.com/MaterializeInc/k8s-eip-operator/pull/474/commits/1b1d20403323796aaad28a08d04affd3bcf0cada)